quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/EventGraphServiceImpl.java

```diff
@@ -2,23 +2,63 @@
 
 import java.util.List;
 
+import org.bson.Document;
+import org.bson.types.ObjectId;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import edu.asu.diging.quadriga.core.data.EventGraphRepository;
 import edu.asu.diging.quadriga.core.model.EventGraph;
 import edu.asu.diging.quadriga.core.service.EventGraphService;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.match;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.newAggregation;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.unwind;
 
+
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.aggregation.CountOperation;
+import org.springframework.data.mongodb.core.aggregation.GroupOperation;
+import org.springframework.data.mongodb.core.aggregation.MatchOperation;
+import org.springframework.data.mongodb.core.aggregation.UnwindOperation;
+import org.springframework.data.mongodb.core.query.Criteria;
 @Service
 public class EventGraphServiceImpl implements EventGraphService {
 
     @Autowired
     private EventGraphRepository repo;
-
+    
+    @Autowired
+    MongoTemplate mongoTemplate;
+    
     @Override
     public void saveEventGraphs(List<EventGraph> events) {
         for (EventGraph event : events) {
             repo.save(event);
         }
     }
+
+    @Override
+    public List<EventGraph> findAllEventGraphsByCollectionId(ObjectId collectionId) {
+        return repo.findByCollectionIdOrderByCreationTimeDesc(collectionId).orElse(null);
+    }
+
+    @Override
+    public long groupEventGraphsBySourceUri(ObjectId collectionId) {
+        UnwindOperation unwind = unwind("context");
+        GroupOperation group = Aggregation.group("context.sourceUri");
+        MatchOperation match= match(Criteria.where("collectionId").is(collectionId));
+        CountOperation count  = Aggregation.count().as("total");
+        Aggregation agg = newAggregation(unwind, match, group, count);
```

Comment:
If I'm not wrong, this aggregation first groups all graphs by sourceuri and then filters by collection id, and then counts all groups not filtered out, right? but if two graphs were created for the same document, the source uri would be the same. This would give the wrong result, wouldn't it?

Reply:
This aggregation pipeline first filters based on collection id, then groups based on sourceUri. count returns the total count of remaining documents in the pipeline. This basically gives the number of distinct sourceUris for a given collection id.

Reply:
yes, but source uris are specific to a text. so three event graphs might have the same source uri if they were all created for the same text. This method however, shoulld return the count of all event graphs in a collelction, right?

also, I"m just noticing, the method should be called `countEventGraphs` or something.